src/main/java/core/basesyntax/ShopServiceImpl.java

```diff
@@ -0,0 +1,29 @@
+package core.basesyntax;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+    private final Map<String, Integer> storage = Storage.getFruitStorage();
```

Comment:
According to the encapsulation principle, the storage map should not be exposed directly. Instead, consider providing methods in the Storage class to manipulate the storage contents.